packages/transaction-controller/src/TransactionController.test.ts

```diff
@@ -3517,4 +3517,55 @@ describe('TransactionController', () => {
       );
     });
   });
+
+  describe('updateTransactionHash', () => {
+    const transactionId = '1';
+    const baseTransaction = {
+      id: transactionId,
+      chainId: toHex(5),
+      status: TransactionStatus.unapproved as const,
+      time: 123456789,
+      txParams: {
+        from: ACCOUNT_MOCK,
+        to: ACCOUNT_2_MOCK,
+      },
+    };
+    const transactionMeta: TransactionMeta = {
+      ...baseTransaction,
+      hash: '1234',
+      history: [{ ...baseTransaction }],
+    };
+    it('updates transaction hash', async () => {
+      const newHash = 'newHash';
+      const controller = newController();
+      controller.state.transactions.push(transactionMeta);
+
+      const updateTransactionSpy = jest.spyOn(controller, 'updateTransaction');
+
+      controller.updateTransactionHash(transactionId, newHash);
+
+      const updatedTransaction = controller.state.transactions.find(
+        (transaction) => transaction.id === transactionId,
+      );
+
+      expect(updateTransactionSpy).toHaveBeenCalledTimes(1);
+      expect(updateTransactionSpy).toHaveBeenCalledWith(
+        updatedTransaction,
+        'TransactionController#updateTransactionHash - Transaction hash updated',
+      );
+      expect(updatedTransaction?.hash).toStrictEqual(newHash);
```

Comment:
This is totally fine but, would it be better if we check controller state rather than asserting spy function, because we expect that change to happen in state with given `transactionId`.

Another nit here is, we will eventually pull down `updateTransaction` method to private and spy on the private methods will not work.

Reply:
Great point! I changed to check the state instead of using the spy function.